dipy/nn/cnn_1d_denoising/cnn_1d_denoising.py

```diff
@@ -0,0 +1,315 @@
+import numpy as np
+import tensorflow as tf
+from tensorflow.keras.layers import Conv1D, Activation
+from tensorflow.keras.models import Model
+from tensorflow.keras.layers import Input
+from sklearn.model_selection import train_test_split
+'''
+Title : Denoising diffusion weighted imaging data using CNN
+=====
+Obtaining tissue microstructure measurements from diffusion weighted imaging (DWI) with multiple,
+high b-values is crucial. However, the high noise levels present in these images can adversely affect
+the accuracy of the microstructural measurements. In this context, we suggest a straightforward denoising technique that
+can be applied to any DWI dataset as long as a low-noise, single-subject dataset is obtained using the same DWI sequence.
+We created a simple 1D-CNN model with five layers, based on the 1D CNN for denoising speech.
+The model consists of two convolutional layers followed by max-pooling layers, and a dense layer.
+The first convolutional layer has 16 one-dimensional filters of size 16, and the second layer has 32 filters of size 8.
+ReLu activation function is applied to both convolutional layers. The max-pooling layer has a kernel
+size of 2 and a stride of 2. 
+The dense layer maps the features extracted from the noisy image to the low-noise reference image.
+
+Reference:
+---------
+Cheng H, Vinci-Booher S, Wang J, Caron B, Wen Q, Newman S, et al.
+(2022) Denoising diffusion weighted imaging data using convolutional neural networks.
+PLoS ONE 17(9): e0274396. https://doi.org/10.1371/journal.pone.0274396
+'''
+class Cnn1DDenoiser:
+    def __init__(self, sig_length, optimizer='adam', loss='mean_squared_error', metrics=('accuracy',),
+         loss_weights=None): 
+        '''
+        Parameters
+        ----------
+        sig_length : Int
+            Length of DWI signal 
+        optimizer : str, optional  
+            Adam (Default) optimization algorithm name
+        loss : str, optional ,mean_squared_error(Default)[fixed]
```

Comment:
remove `,mean_squared_error(Default)[fixed]` or move it below with `select loss...`